.github/workflows/maven-verify.yml

```diff
@@ -15,19 +15,49 @@
 # specific language governing permissions and limitations
 # under the License.
 
-name: Verify
+name: GitHub CI
 
 on:
   push:
     branches-ignore:
       - dependabot/**
   pull_request:
 
+
 jobs:
   build:
-    name: Verify
-    uses: apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v1
-    with:
-      ff-goal: 'install'
-      verify-goal: 'install -P run-its'
-      verify-fail-fast: false
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        java: [8, 11, 17]
+        jdk: [temurin]
+      fail-fast: false
+
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 120
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v2.4.0
+        with:
+          distribution: ${{ matrix.jdk }}
+          java-version: ${{ matrix.java }}
+          cache: 'maven'
+
+      - name: Build with Maven
+        run: mvn clean install -e -B -V -nsu --no-transfer-progress -P run-its -Dfailsafe-integration-test-port=8083
+
+      - name: Upload artifact surefire-its
+        uses: actions/upload-artifact@v2-preview
+        if: failure()
+        with:
+          name: ${{ matrix.os }}-surefire-its
+          path: |
+            surefire-its/target/*/log.txt
+            surefire-its/target/**/surefire-reports/*
+            surefire-its/target/**/failsafe-reports/*
+            !surefire-its/target/failsafe-reports
```

Comment:
Don't do it in this way
it is step backward

Reply:
The commit message is `investigating`. The commit does not mean anything for this Jira issue. We are discussing the timeout in the Slack/INFRA channel and I need to get the archive of `surefire-its/target` with all the dump files and logs. This was happening one year ago and I came over this issue by splitting the test set in two. Currently I am waiting for the build to complete and I hope the archive files would appear in the attachment. Then I want to go inside and analyse the logs and test content.

Reply:
@slawekjaranowski
The build runs every configuration twice, means `push` and `pull_request`. This is the waste and I also could not avoid it in my scripts. Do you know how to do it? This would save the h/w resources in favor of others.

Reply:
That is because we have two events:
- push to branches
- pull_request 

I try to discover how prevent it

Reply:
by the way attached artifacts has size > 1GB ...

Reply:
@slawekjaranowski
This change does not exit in this branch any longer.